mods/weather/api.lua

```diff
@@ -0,0 +1,32 @@
+local CYCLE = 8 -- Time period of cyclic clouds update in seconds
+
+weather = {}
+
+-- default implementation is empty
+function weather.get(player)
+	return {}
```

Comment:
Reading this is a bit misleading: This function is overwritten in `init.lua`. So the first thing a reader is going to think is "wait, doesn't this disable weather?"

Why define it here at all? I'd just leave a comment here along the lines of "`weather.get` is defined in `init.lua`". `do_update` is not called at load time, so this should be fine.

Or if you do want to leave a function here, to make the "dependency" explicit, why not something like `error("this should be overridden in init.lua")`?

Though the ideal solution would probably be to move the appropriate code from `init.lua` into `api.lua`, or even something like a separate file `default.lua` or similar, such that you could `dofile` it from here and have it be loaded in time as to not confuse a reader where `weather.get` comes from.

Reply:
init.lua redefines this, but only conditionally. For example it doesn't when `enable_weather` is false.

Reply:
Right. A comment explaining this may be a good idea. The early return is easy to miss.